### PR TITLE
Support to run ONNX Upsample operator (mode=nearest) in Caffe2

### PR DESCRIPTION
**Pull request by varunjain99**
Added to the python and C++ backend in pytorch/caffe2/python/onnx and pytorch/caffe2/onnx. Upsample tests should now pass

### CODE REVIEW

**Review by bddppq — COMMENTED** [review]
Nice work!

**Comment by bddppq** [comment]
The submodule third_party/onnx should be changed

File: caffe2/python/onnx/backend.py [hunk 1]
```diff
@@ -45,6 +45,9 @@
 
 import warnings
 
+def almost_equal(a, b):
+    return (abs(f1 - f2) <= 1e-15)
```
Comment (bddppq) [inline caffe2/python/onnx/backend.py]: Use np.isclose

**Comment by bddppq** [comment]
@onnxbot retest this please

**Comment by bddppq** [comment]
There are compilation errors:
```
/var/lib/jenkins/workspace/caffe2/onnx/backend.cc:891:33: error: no match for 'operator[]' (operand types are 'google::protobuf::RepeatedField<float>' and 'int')
00:07:16    } else if (!AlmostEqual(scales[0], 1) || !AlmostEqual(scales[1], 1))  {
00:07:16                                  ^
00:07:16 /var/lib/jenkins/workspace/caffe2/onnx/backend.cc:891:63: error: no match for 'operator[]' (operand types are 'google::protobuf::RepeatedField<float>' and 'int')
00:07:16    } else if (!AlmostEqual(scales[0], 1) || !AlmostEqual(scales[1], 1))  {
00:07:16                                                                ^
00:07:16 /var/lib/jenkins/workspace/caffe2/onnx/backend.cc:900:26: error: no match for 'operator[]' (operand types are 'google::protobuf::RepeatedField<float>' and 'int')
00:07:16    c2_height->set_f(scales[2]);
```

**Review by bddppq — APPROVED** [review]
Great work!

**Comment by albu** [comment]
This commit breaks exporting models with UpSampling2D layer from pytorch 0.4.0 (and also master) with onnx from master. Reverting it helped with this issue.

**Comment by bddppq** [comment]
@albu Could you elaborate? This pr only touched the onnx -> caffe2 importing part, how does it affect the exporter?

**Comment by albu** [comment]
Sorry, I mean onnx -> caffe2 importing.

It always throws
`CAFFE_THROW("The scales argument should have size 4");`
and in onnx definition we already have `height_scale` and `width_scale` fields so no need to redefine them

**Pull request merged**